### ISSUE TITLE
Use BytesIO() instead of MockFile() in xml_requests.py

### ISSUE THREAD

**Issue by harshavardhana** [minio/minio-py#215]
To avoid future bugs it is beneficial to use io.BytesIO()

**Comment by fkautz** [comment]
BytesIO provides only read, not write. This method merely collects bytes into a byte array. All encoding should already be performed before this write() is called.

Any other suggestions? Otherwise, this should be safe to close.

**Comment by harshavardhana** [comment]
> BytesIO provides only read, not write

I can write to BytesIO()

```python
>>> from io import BytesIO
>>> value = BytesIO()
>>> value.write("hello")
5L
>>> value.getvalue()
'hello'
>>>
```

So it is clearly possible to write to BytesIO()

**Comment by harshavardhana** [comment]
For example the following change works fine.. 

~~~
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from io import BytesIO
 from xml.etree import ElementTree
 
 __author__ = 'minio'
@@ -20,16 +21,9 @@ def bucket_constraint(region):
     root = ElementTree.Element('CreateBucketConfiguration', {'xmlns': 'http://s3.amazonaws.com/doc/2006-03-01/'})
     location_constraint = ElementTree.SubElement(root, 'LocationConstraint')
     location_constraint.text = region
-    data = []
-    mock_file = MockFile()
-    mock_file.write = data.append
-    ElementTree.ElementTree(root).write(mock_file, encoding=None, xml_declaration=False)
-    return b''.join(data)
-
-
-class MockFile(object):
-    pass
-
+    data = BytesIO()
+    ElementTree.ElementTree(root).write(data, encoding=None, xml_declaration=False)
+    return data.getvalue()
 
 def generate_complete_multipart_upload(etags):
     root = ElementTree.Element('CompleteMultipartUpload', {'xmlns': 'http://s3.amazonaws.com/doc/2006-03-01/'})
@@ -40,8 +34,6 @@ def generate_complete_multipart_upload(etags):
         part_number.text = str(i + 1)
         etag = ElementTree.SubElement(part, 'ETag')
         etag.text = etags[i]
-    data = []
-    mock_file = MockFile()
-    mock_file.write = data.append
-    ElementTree.ElementTree(root).write(mock_file, encoding=None, xml_declaration=False)
-    return b''.join(data)
+    data = BytesIO()
+    ElementTree.ElementTree(root).write(data, encoding=None, xml_declaration=False)
+    return data.getvalue()
~~~ 

All tests run okay. even integration tests run fine..

**Comment by fkautz** [comment]
Ah, mistake I ran into was trying to use read() instead of getvalue. I'll reproduce and resubmit. Next time feel free to submit your change. :)

**Comment by fkautz** [comment]
Fixed with #217
